code/game/objects/structures/barsign.dm

```diff
@@ -41,363 +53,729 @@
+			new /obj/item/shard(get_turf(user))
+		else
+			to_chat(user, "<span class='notice'>You pull the glass screen out from [src].</span>")
+			new /obj/item/stack/sheet/glass(get_turf(user), 2)
+		build_stage = BARSIGN_WIRED
+		update_icon()
+		add_fingerprint(user)
+	I.play_tool_sound(user, I.tool_volume)
 
-/obj/structure/sign/barsign/emag_act(mob/user)
-	if(broken || emagged)
-		to_chat(user, "<span class='warning'>Nothing interesting happens!</span>")
+/obj/machinery/barsign/wirecutter_act(mob/living/user, obj/item/I)
+	if(user.a_intent != INTENT_HELP)
+		return
+	if(build_stage != BARSIGN_WIRED)
+		return
+	. = TRUE
+	if(I.tool_behaviour != TOOL_WIRECUTTER)
+		return
+	if(!I.use_tool(src, user, 0, volume = 0))
+		return
+	if(!(flags & NODECONSTRUCT))
+		if(stat & EMPED)
+			to_chat(user, "<span class='notice'>You remove the burnt wires out from [src].</span>")
+		else
+			to_chat(user, "<span class='notice'>You cut the wires out from [src].</span>")
+			new /obj/item/stack/cable_coil(get_turf(user), 5)
+		build_stage = BARSIGN_CIRCUIT
+		update_icon()
+		power_state = NO_POWER_USE
+		I.play_tool_sound(user, I.tool_volume)
+		add_fingerprint(user)
+
+/obj/machinery/barsign/emag_act(mob/user)
+	if(stat & (BROKEN|NOPOWER|EMPED))
+		to_chat(user, "<span class='warning'>[src] cannot be taken over right now!</span>")
+		return
+	if(emagged)
+		to_chat(user, "<span class='warning'>[src] is already taken over!</span>")
 		return
 	to_chat(user, "<span class='notice'>You emag the barsign. Takeover in progress...</span>")
+	add_fingerprint(user)
 	addtimer(CALLBACK(src, PROC_REF(post_emag)), 10 SECONDS)
 	return TRUE
 
-/obj/structure/sign/barsign/proc/post_emag()
-	if(broken || emagged)
+/obj/machinery/barsign/proc/post_emag()
+	if(stat & (BROKEN|NOPOWER|EMPED))
 		return
-	set_sign(new /datum/barsign/hiddensigns/syndibarsign)
 	emagged = TRUE
-	req_access = list(ACCESS_SYNDICATE)
+	set_random_sign()
+	req_access = list()
+
+/obj/machinery/barsign/emp_act(severity)
+	if(build_stage >= BARSIGN_WIRED)
+		if(!(stat & EMPED))
+			stat |= EMPED
+			turn_on()
+			set_sign(new /datum/barsign/hiddensigns/empbarsign)
+			playsound(loc, 'sound/effects/sparks4.ogg', 60, TRUE)
+			return ..()
 
-/obj/structure/sign/barsign/proc/pick_sign()
-	var/picked_name = tgui_input_list(usr, "Available Signage", "Bar Sign", barsigns)
-	if(!picked_name)
+/obj/machinery/barsign/power_change()
+	. = ..()
+	if(. && (stat & NOPOWER))
+		turn_off()
+
+/obj/machinery/barsign/deconstruct(disassembled = FALSE)
+	if(disassembled)
+		new /obj/item/stack/sheet/metal(drop_location(), 4)
+	else
+		new /obj/item/stack/sheet/metal(drop_location(), 2)
+		if(build_stage >= BARSIGN_WIRED && !(stat & EMPED))
+			new /obj/item/stack/cable_coil(drop_location(), 3)
+		if(build_stage >= BARSIGN_COMPLETE)
+			new /obj/item/shard(drop_location())
+	qdel(src)
+
+
+/datum/stack_recipe/barsign_frame
+
+/datum/stack_recipe/barsign_frame/try_build(mob/user, obj/item/stack/S, multiplier)
```

Comment:
Why make this a ```datum/stack_recipe``` instead of a ```/obj/item/mounted/frame``` with some adjustments?

Reply:
Since the barsign is two tiles it would need a custom sprite (air, fire alarms, etc just use the wall sprite), and I determined that a "good" sprite would be near impossible irrelevant of my skill (or lack thereof) in sprite creation.  A barsign frame is after all just a two-color rectangle, so it wouldn't ever look 'good', or even be recognizable as a barsign frame.  It would also look a bit strange when dropped on the floor due to the size difference between the sprite and how big it 'should' be.  Because of this, I decided that it should just be constructed directly onto the wall.

Reply:
mounted/frame would be the way to go if you had a good sprite for the frame you think? I could play with it and see if anything decent comes out.

Reply:
If you think you can create one, sure, but I don't see how it could be anything other than a non-descript rectangle or brick.